plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp

```diff
@@ -625,13 +642,11 @@ class read_write {
      using index_type = chain::index256_index;
      static auto function() {
         return [](const input_type& v) {
-            chain::key256_t k;
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wstrict-aliasing"
-            k[0] = ((uint128_t *)&v._hash)[0]; //0-127
-            k[1] = ((uint128_t *)&v._hash)[1]; //127-256
-#pragma GCC diagnostic pop
-            return k;
+            // The input is in big endian, i.e. f58262c8005bb64b8f99ec6083faf050c502d099d9929ae37ffed2fe1bb954fb
+            // fixed_bytes will convert the input to array of 2 uint128_t in little endian, i.e. 50f0fa8360ec998f4bb65b00c86282f5 fb54b91bfed2fe7fe39a92d999d002c5
+            // which is the format used by secondary index
+            fixed_bytes<32> fb(*reinterpret_cast<const std::array<const uint8_t, 32>*>(v.data())); 
```

Comment:
Since `std::array` is a struct, this is a strict aliasing violation. It's probably safest to memcpy from `v.data()` into a local array, then pass that array to the `fixed_bytes` constructor.